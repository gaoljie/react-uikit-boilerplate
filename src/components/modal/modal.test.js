import React from "react";
import "jest-styled-components";
import { mount } from "enzyme";

import styled from "styled-components";
import Modal from "./index";

const Container = styled.div`
  width: 400px;
  height: 400px;
  background-color: yellow;
`;

const Container2 = styled.div`
  width: 400px;
  height: 400px;
  background: yellow;
`;

const ContainerWithoutBg = styled.div`
  width: 400px;
  height: 400px;
`;

describe("Test Input", () => {
  window.scrollTo = jest.fn();

  it("Matches the snapshot", () => {
    const wrapper = mount(
      <Modal visible>
        <Container>Container</Container>
      </Modal>
    );
    expect(wrapper).toMatchSnapshot();
  });

  it("Render Prop", () => {
    const wrapper = mount(
      <Modal>
        <Container>Text</Container>
      </Modal>
    );

    expect(wrapper).toHaveStyleRule("display", "none");
    expect(wrapper.find(Container)).toHaveStyleRule(
      "background-color",
      "yellow"
    );

    wrapper.setProps({ visible: true });

    expect(wrapper).not.toHaveStyleRule("display");
    expect(wrapper.text()).toEqual("Text");

    const evt = { stopPropagation: jest.fn() };

    wrapper.find(Container).simulate("click", evt);
    expect(evt.stopPropagation).toHaveBeenCalledTimes(1);
  });

  it("Styling", () => {
    const wrapper = mount(
      <Modal visible={false}>
        <ContainerWithoutBg>Text</ContainerWithoutBg>
      </Modal>
    );

    expect(wrapper.find(ContainerWithoutBg)).toHaveStyleRule(
      "background-color",
      "white"
    );
  });

  it("Styling2", () => {
    const wrapper = mount(
      <Modal visible={false}>
        <Container2>Text</Container2>
      </Modal>
    );

    expect(wrapper.find(Container2)).toHaveStyleRule("background", "yellow");
  });
});
