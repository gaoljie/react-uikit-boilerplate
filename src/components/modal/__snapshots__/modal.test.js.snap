// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Input Matches the snapshot 1`] = `
.c0 {
  position: fixed;
  top: 0;
  bottom: 0;
  left: 0;
  right: 0;
  z-index: 1000;
  background-color: rgba(0,0,0,0.3);
}

.c1 {
  position: absolute;
  top: 50%;
  left: 50%;
  -webkit-transform: translate(-50%,-50%);
  -ms-transform: translate(-50%,-50%);
  transform: translate(-50%,-50%);
  overflow: auto;
}

.c2 {
  width: 400px;
  height: 400px;
  background-color: yellow;
}

<Modal
  visible={true}
>
  <styled.div
    visible={true}
  >
    <StyledComponent
      forwardedComponent={
        Object {
          "$$typeof": Symbol(react.forward_ref),
          "attrs": Array [],
          "componentStyle": ComponentStyle {
            "componentId": "sc-bdVaJa",
            "isStatic": false,
            "lastClassName": "c0",
            "rules": Array [
              "
  ",
              [Function],
              ";
  position: fixed;
  top: 0;
  bottom: 0;
  left: 0;
  right: 0;
  z-index: 1000;
  background-color: rgba(0, 0, 0, 0.3);
",
            ],
          },
          "displayName": "styled.div",
          "foldedComponentIds": Array [],
          "render": [Function],
          "styledComponentId": "sc-bdVaJa",
          "target": "div",
          "toString": [Function],
          "warnTooManyClasses": [Function],
          "withComponent": [Function],
        }
      }
      forwardedRef={null}
      visible={true}
    >
      <div
        className="c0"
      >
        <styled.div
          as={
            Object {
              "$$typeof": Symbol(react.forward_ref),
              "attrs": Array [],
              "componentStyle": ComponentStyle {
                "componentId": "sc-htpNat",
                "isStatic": false,
                "lastClassName": "c2",
                "rules": Array [
                  "
  width: 400px;
  height: 400px;
  background-color: yellow;
",
                ],
              },
              "displayName": "styled.div",
              "foldedComponentIds": Array [],
              "render": [Function],
              "styledComponentId": "sc-htpNat",
              "target": "div",
              "toString": [Function],
              "warnTooManyClasses": [Function],
              "withComponent": [Function],
            }
          }
          hasBgAttr={true}
          onClick={[Function]}
        >
          <StyledComponent
            as={
              Object {
                "$$typeof": Symbol(react.forward_ref),
                "attrs": Array [],
                "componentStyle": ComponentStyle {
                  "componentId": "sc-htpNat",
                  "isStatic": false,
                  "lastClassName": "c2",
                  "rules": Array [
                    "
  width: 400px;
  height: 400px;
  background-color: yellow;
",
                  ],
                },
                "displayName": "styled.div",
                "foldedComponentIds": Array [],
                "render": [Function],
                "styledComponentId": "sc-htpNat",
                "target": "div",
                "toString": [Function],
                "warnTooManyClasses": [Function],
                "withComponent": [Function],
              }
            }
            forwardedComponent={
              Object {
                "$$typeof": Symbol(react.forward_ref),
                "attrs": Array [],
                "componentStyle": ComponentStyle {
                  "componentId": "sc-bwzfXH",
                  "isStatic": false,
                  "lastClassName": "c1",
                  "rules": Array [
                    "
  position: absolute;
  top: 50%;
  left: 50%;
  transform: translate(-50%, -50%);
  overflow: auto;
  ",
                    [Function],
                    "
",
                  ],
                },
                "displayName": "styled.div",
                "foldedComponentIds": Array [],
                "render": [Function],
                "styledComponentId": "sc-bwzfXH",
                "target": "div",
                "toString": [Function],
                "warnTooManyClasses": [Function],
                "withComponent": [Function],
              }
            }
            forwardedRef={null}
            hasBgAttr={true}
            onClick={[Function]}
          >
            <styled.div
              className="c1"
              hasBgAttr={true}
              onClick={[Function]}
            >
              <StyledComponent
                className="c1"
                forwardedComponent={
                  Object {
                    "$$typeof": Symbol(react.forward_ref),
                    "attrs": Array [],
                    "componentStyle": ComponentStyle {
                      "componentId": "sc-htpNat",
                      "isStatic": false,
                      "lastClassName": "c2",
                      "rules": Array [
                        "
  width: 400px;
  height: 400px;
  background-color: yellow;
",
                      ],
                    },
                    "displayName": "styled.div",
                    "foldedComponentIds": Array [],
                    "render": [Function],
                    "styledComponentId": "sc-htpNat",
                    "target": "div",
                    "toString": [Function],
                    "warnTooManyClasses": [Function],
                    "withComponent": [Function],
                  }
                }
                forwardedRef={null}
                hasBgAttr={true}
                onClick={[Function]}
              >
                <div
                  className="c2 c1"
                  onClick={[Function]}
                >
                  Container
                </div>
              </StyledComponent>
            </styled.div>
          </StyledComponent>
        </styled.div>
      </div>
    </StyledComponent>
  </styled.div>
</Modal>
`;
